anta/catalog.py

```diff
@@ -0,0 +1,152 @@
+                            description: "Test with overwritten description"
+                            custom_field: "Test run by John Doe"
+
+        Args:
+            test_catalog: Python dictionary representing the test catalog YAML file
+
+        """
+        # pylint: disable=broad-exception-raised
+        if not test_catalog:
+            return
+
+        for key, value in test_catalog.items():
+            # Required to manage iteration within a tests module
+            if package is not None:
+                key = ".".join([package, key])
+            try:
+                module = importlib.import_module(f"{key}")
+            except ModuleNotFoundError:
+                logger.critical(f"No test module named '{key}'")
+                raise
+
+            if isinstance(value, list):
+                # This is a list of tests
+                for test in value:
+                    for test_name, inputs in test.items():
+                        # A test must be a subclass of AntaTest as defined in the Python module
+                        try:
+                            test = getattr(module, test_name)
+                        except AttributeError:
+                            logger.critical(f"Wrong test name '{test_name}' in '{module.__name__}'")
+                            raise
+                        if not issubclass(test, AntaTest):
+                            logger.critical(f"'{test.__module__}.{test.__name__}' is not an AntaTest subclass")
+                            raise Exception()
+                        # Test inputs can be either None or a dictionary
+                        if inputs is None or isinstance(inputs, dict):
+                            self.tests.append((cast(AntaTest, test), cast(AntaTest.Input, inputs)))
+                        else:
+                            logger.critical(f"'{test.__module__}.{test.__name__}' inputs must be a dictionary")
+                            raise Exception()
+            if isinstance(value, dict):
+                # This is an inner Python module
+                self.parse_catalog(value, package=module.__name__)
+
+    def check(self: AntaCatalog) -> ResultManager:
+        """
+        TODO - for now a test requires a device but this may be revisited in the future
+        """
+        # Mock device
+        mock_device = AsyncEOSDevice(name="mock", host="127.0.0.1", username="mock", password="mock")
+
+        manager = ResultManager()
+        # Instantiate each test to verify the Inputs are correct
+        for test_class, test_inputs in self.tests:
+            # TODO - this is the same code with typing as in runner.py but somehow mypy complains that test_class
+            # ot type AntaTest is not callable
+            test_instance = test_class(device=mock_device, inputs=test_inputs)  # type: ignore[operator]
+            manager.add_test_result(test_instance.result)
+        return manager
```

Comment:
this is crap use https://docs.pydantic.dev/latest/concepts/models/#helper-functions `model_validate`